jujugui/static/gui/src/app/user/user.js

```diff
@@ -0,0 +1,88 @@
+/*
+ This file is part of the Juju GUI, which lets users view and manage Juju
+ environments within a graphical interface (https://launchpad.net/juju-gui).
+ Copyright (C) 2017 Canonical Ltd.
+
+ This program is free software: you can redistribute it and/or modify it under
+ the terms of the GNU Affero General Public License version 3, as published by
+ the Free Software Foundation.
+
+ This program is distributed in the hope that it will be useful, but WITHOUT
+ ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+ SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+ General Public License for more details.
+
+ You should have received a copy of the GNU Affero General Public License along
+ with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+'use strict';
+
+if (typeof this.jujugui === 'undefined') {
+  this.jujugui = {};
+}
+
+/** Class representing a user's authorizations in the GUI **/
+const User = class User {
+
+  constructor(cfg = {}) {
```

Comment:
I think it's quite likely that as we add the expiration stuff and the other credentials to this object cfg is going to grow. If it doesn't it's easy to change later.